modules/xfeatures2d/src/daisy.cpp

```diff
@@ -0,0 +1,2057 @@
+    // and we warp this grid with H and compute the descriptor on this warped
+    // grid; returns null/false if centers falls outside the image; allocate
+    // 'descriptor' memory first. descriptor is NOT normalized.
+    inline bool get_unnormalized_descriptor( double y, double x, int orientation, double* H, float* descriptor );
+
+    // compute the smoothed gradient layers.
+    inline void compute_smoothed_gradient_layers();
+
+    // does not use interpolation while computing the histogram.
+    inline void ni_get_histogram( float* histogram, int y, int x, int shift, float* hcube );
+
+    // returns the interpolated histogram: picks either bi_get_histogram or
+    // ti_get_histogram depending on 'shift'
+    inline void i_get_histogram( float* histogram, double y, double x, double shift, float* cube );
+
+    // records the histogram that is computed by bilinear interpolation
+    // regarding the shift in the spatial coordinates. hcube is the
+    // histogram cube for a constant smoothness level.
+    inline void bi_get_histogram( float* descriptor, double y, double x, int shift, float* hcube );
+
+    // records the histogram that is computed by trilinear interpolation
+    // regarding the shift in layers and spatial coordinates. hcube is the
+    // histogram cube for a constant smoothness level.
+    inline void ti_get_histogram( float* descriptor, double y, double x, double shift, float* hcube );
+
+    // uses interpolation, for no interpolation call ni_get_descriptor. see also get_descriptor
+    inline void i_get_descriptor( double y, double x, int orientation, float* descriptor );
+
+    // does not use interpolation. for w/interpolation, call i_get_descriptor. see also get_descriptor
+    inline void ni_get_descriptor( double y, double x, int orientation, float* descriptor );
+
+    // uses interpolation for no interpolation call ni_get_descriptor. see also get_descriptor
+    inline bool i_get_descriptor( double y, double x, int orientation, double* H, float* descriptor );
+
+    // does not use interpolation. for w/interpolation, call i_get_descriptor. see also get_descriptor
+    inline bool ni_get_descriptor( double y, double x, int orientation, double* H, float* descriptor );
+
+    // creates a 1D gaussian filter with N(mean,sigma).
+    inline void gaussian_1d( float* fltr, int fsz, float sigma, float mean )
+    {
+      CV_Assert(fltr != NULL);
+      int sz = (fsz-1)/2;
+      int counter=-1;
+      float sum = 0.0;
+      float v = 2*sigma*sigma;
+      for( int x=-sz; x<=sz; x++ )
+      {
+         counter++;
+         fltr[counter] = exp((-((float)x-mean)*((float)x-mean))/v);
+         sum += fltr[counter];
+      }
+
+      if( sum != 0 )
+      {
+         for( int x=0; x<fsz; x++ )
+            fltr[x] /= sum;
+      }
+    }
+
+    inline void conv_horizontal( float* image, int h, int w, float* kernel, int ksize )
+    {
+      CvMat cvI; cvInitMatHeader(&cvI, h, w, CV_32FC1, (float*)image);
```

Comment:
[DONE]
* I used now filter2D twice for H,V steps. Was not able to use the nice getGaussianKernel() because it differs in results at 10e-6 and daisy produced different results ! Also was not able to use sepFilter2D(), but filter2D worked perfect using BORDER_REPLIACATE and not default one. 
* Converted to opencv, and we still produce 1:1 at bit level same numerical results.